### PR TITLE
Introduce non-static class

### PR DESCRIPTION
**Pull request by markusheiden**
It is a bit suboptimal for future developments that all methods are static. So I made them non-static.

### CODE REVIEW

File: src/main/java/com/javaetmoi/core/persistence/hibernate/Hydrator.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    /**
+     * Factory for {@link EntityManagerFactory} or {@link SessionFactory}.
+     *
+     * @param entityManagerFactory
+     *            Open {@link EntityManagerFactory}.
+     */
+    public static Hydrator hydrator(EntityManagerFactory entityManagerFactory) {
+        return new HydratorImpl(entityManagerFactory);
```
Comment (arey) [inline src/main/java/com/javaetmoi/core/persistence/hibernate/Hydrator.java:32]: It always concerns me a little when an interface knows its implementation class.
Do you know if this pattern is adopted by the java community?

Comment (markusheiden) [reply src/main/java/com/javaetmoi/core/persistence/hibernate/Hydrator.java:32]: I saw it a few times, but I don't know whether it is adopted by the community.

I did this to hide the implementation and to reduce the number of classes. Alternatives would be:
* A `HydratorFactory` containing the static methods.
* Make `HydratorImpl` public.

**Comment by markusheiden** [comment]
It's Markus ;-)

I will add some docs.

File: src/main/java/com/javaetmoi/core/persistence/hibernate/Hydrator.java (hunk 1 above)
Comment (arey) [reply src/main/java/com/javaetmoi/core/persistence/hibernate/Hydrator.java:32]: Markus, I propose to keep the factory method in the `Hydrator` interface like you've done. I'm beginning to like it :-D

**Pull request merged**